File: sdk/web/component/sqljson/controller/EmailController.js
var EmailController = function EmailController(cfg)
{
    let ctrl = {};
    
    ctrl.ReadEmail = function (onResponse)
    {
        cfg.Connector.Submit(format, "ardvro/component/sqljson/Common/ReadEmail", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.SendEmail = function (to, subject, content, onResponse)
    {
        let args = [to, subject, content];
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Email/SendEmail", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.SendEmailWithAttachment = function (to, subject, content, filename, filecontent, onResponse)
    {
        let args = [to, subject, content, filename, filecontent];
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Email/SendEmailWithAttachment", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.ContactUs = function (fromEmail, fromName, subject, content, onResponse)
    {
        content = "".concat(fromName, " ", fromEmail, " <br><br>", content);
        let args = [fromEmail, fromName, subject, content];
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Email/ContactUs", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    return ctrl;
};

File: sdk/.net/ardvro.sdk.net/component/sqljson/controller/EmailController.cs
namespace ardvro.sdk.net.component.sqljson.controller;

using ardvro.core.ext.connection;
using ardvro.core.lib.common;

public class EmailController
{
    private IConnection _conection;

    public EmailController(IConnection connection)
    {
        _conection = connection;
    }

    public void SendEmail(String to, String subject, String content, Action<int> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Email/SendEmail";
        _conection.Submit(new Object[]{ to, subject, content }, functionName, onResponse);
    }

    public void SendEmailWithAttachment(String to, String subject, String content, String filename, String filecontent, Action<int> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Email/SendEmailWithAttachment";
        _conection.Submit(new Object[]{ to, subject, content, filename, filecontent }, functionName, onResponse);
    }

    public void ContactUs(String fromEmail, String fromName, String subject, String content, Action<int> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Email/SendEmailWithAttachment";
        _conection.Submit(new Object[]{ fromEmail, fromName, subject, content }, functionName, onResponse);
    }

    public void ReadEmail(Action<List<Flexible>> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Email/ReadEmail";
        _conection.Submit(new Object[]{  }, functionName, onResponse);
    }

}
